mysql-test/t/percona_binlog_utils_udf.test

```diff
@@ -0,0 +1,312 @@
+--source include/have_udf.inc
+--source include/have_grep.inc
+#
+# Create functions from binlog_utils_udf
+#
+
+--disable_query_log
+SET @path_separator = '/';
+if(`SELECT CONVERT(@@version_compile_os USING latin1) IN ("Win32", "Win64", "Windows") = 1`)
+{
+  SET @path_separator = '\\';
+}
+--enable_query_log
+
+--echo
+--echo *** creating a table that will be used to generate binlog events
+CREATE TABLE t1(
+  id BIGINT UNSIGNED NOT NULL AUTO_INCREMENT,
+  PRIMARY KEY(id)
+) ENGINE=InnoDB;
+
+--echo
+--echo *** creating a temporary table that will store recorded binlog names and GTIDs
+CREATE TEMPORARY TABLE captured_gtid(
+  stage BIGINT UNSIGNED NOT NULL,
+  substage BIGINT UNSIGNED NOT NULL,
+  binlog_name VARCHAR(64) NOT NULL,
+  gtid_value VARCHAR(64) NOT NULL,
+  PRIMARY KEY(stage, substage)
+) ENGINE=InnoDB;
+
+# Flush binary logs so that previous "CREATE TABLE" statement won't affect GTID
+# recording logic.
+FLUSH BINARY LOGS;
+
+--echo
+--echo *** generating several binlogs
+--disable_query_log
+--disable_result_log
+--let $number_of_stages = 16
+--let $number_of_substages = 16
+
+--let $stage = 1
+while($stage <= $number_of_stages)
+{
+  --let $binlog_file_name = query_get_value(SHOW MASTER STATUS, File, 1)
+
+  --let $substage = 1
+  while($substage <= $number_of_substages)
+  {
+    SET @saved_gtid_executed = @@global.gtid_executed;
+    START TRANSACTION;
+    --let $i = 1
+    while ($i <= $substage)
+    {
+      INSERT INTO t1 VALUES(DEFAULT);
+      --inc $i
+    }
+    COMMIT;
+    eval INSERT INTO captured_gtid VALUES($stage, $substage, '$binlog_file_name', GTID_SUBTRACT(@@global.gtid_executed, @saved_gtid_executed));
```

Comment:
Can be ignored. I prefer to use `--eval` instead of `eval ... ;` to better distinguish `mysqltest` commands from SQL statements.

Reply:
@inikep To be honest a also prefer `--command ...` syntax rather than `command ...;`.
However, for executing SQL statements that require `$var` substitutions I usually use the latter because this syntax supports multiline statements (and very often this is the case).
So, this is actually where I bend the rule deliberately to avoid situations where I will be using `--eval ...` for single line and `eval ...;` for multiline statements - just to make everything uniform.